inference-engine/src/vpu/graph_transformer/src/stages/loop_start.cpp

```diff
@@ -70,8 +71,53 @@ class LoopStart : public StageNode {
 
 }  // namespace
 
+namespace {
+
+class LoopDynamicIterNumber : public StageNode {
+public:
+    using StageNode::StageNode;
+
+protected:
+    StagePtr cloneImpl() const override {
+        return std::make_shared<LoopDynamicIterNumber>(*this);
+    }
+
+    // TODO: should this method be empty?
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder>& orderInfo) override {
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement>& stridesInfo) override {
+    }
+
+    void finalizeDataLayoutImpl() override {
+    }
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport>& /*batchInfo*/) override {
+    }
+
+    void initialCheckImpl() const override {
+    }
+
+    void serializeParamsImpl(BlobSerializer& serializer) const override {
+        const auto numDims = inputEdge(0)->input()->desc().numDims();
+        const auto batchDimInd = numDims - 1 - dimToIeInd(Dim::N, numDims);
+        serializer.append(static_cast<uint32_t>(batchDimInd));
```

Comment:
Would it be always last (MDK) or first (IE) dimension?

Reply:
I guess we should pass batch index since batch can be in any dimension potentially.